UnitTest/TestComponentCSharp_Tests.cs

```diff
@@ -53,6 +53,150 @@ public void TestEmptyBufferToArray()
         }
 
         [Fact]
+        public void TestBufferCheck1()
+        {
+            IBuffer buf = new Windows.Storage.Streams.Buffer(3);
+            byte[] arr = new byte[] { 0x01, 0x02, 0x03 };
+            // Do something with buffer and arr
+            // CopyTo(this byte[] source, int sourceIndex, IBuffer destination, uint destinationIndex, int count)
+            arr.CopyTo(3, buf, 3, 0);
+        }
+        [Fact(Skip="errors properly")]
+        public void TestBufferCheck11()
+        {
+            IBuffer buf = new Windows.Storage.Streams.Buffer(3);
+            byte[] arr = new byte[] { 0x01, 0x02, 0x03 };
+            arr.CopyTo(3, buf, 3, 1); // errors 

```

Comment:
This case is interesting in comparison to the above, as I anticipated it to go through but an exception did get thrown.

Reply:
why do you expect this to go through? you are attempting to copy into range `[3;4)` using `[offset;ending)` notation, and 4 is clearly beyond array length 3.

Reply:
I don't expect it to go through, I wanted to make sure the change in condition checks didn't allow this to go through